app/src/main/java/com/keylesspalace/tusky/components/notifications/NotificationFetcher.kt

```diff
@@ -1,64 +1,138 @@
 package com.keylesspalace.tusky.components.notifications
 
+import android.app.NotificationManager
 import android.content.Context
+import android.os.Looper
 import android.util.Log
+import com.keylesspalace.tusky.components.notifications.NotificationHelper.filterNotification
 import com.keylesspalace.tusky.db.AccountEntity
 import com.keylesspalace.tusky.db.AccountManager
 import com.keylesspalace.tusky.entity.Marker
 import com.keylesspalace.tusky.entity.Notification
 import com.keylesspalace.tusky.network.MastodonApi
-import com.keylesspalace.tusky.util.isLessThan
 import javax.inject.Inject
+import kotlin.math.min
 
 class NotificationFetcher @Inject constructor(
     private val mastodonApi: MastodonApi,
     private val accountManager: AccountManager,
     private val context: Context
 ) {
     fun fetchAndShow() {
+        // This should never be run on the main thread (see
+        // https://developer.android.com/guide/background/persistent/threading/worker). Belt-and-braces
+        // check to make sure.
+        if (Looper.getMainLooper().isCurrentThread) {
+            throw IllegalStateException("fetching notifications on the main thread")
+        }
```

Comment:
That's not what I see.

I just checked this by doing the following:

In `MainActivity.onCreate()` I put the following code immediately after the `super.onCreate(savedInstanceState)` call:

```kotlin
NotificationFetcher(mastodonApi, accountManager, this).fetchAndShow()
```

and commented out the `throw IllegalStateException` on here.

Rebuilt and restarted the app and the notifications were fetched.

Then uncommented the `throw IllegalStateException`, repeated the test, and the exception was thrown as expected.

Looking in to this some more the check that I've got here can be replaced with an `@WorkerThread` annotation on the class. That won't be a runtime error, but it will cause the lint checks to fail, and since that will blocks PRs I think that's OK.

Reply:
yes an annotation is a better idea